src/Application.php

```diff
@@ -74,11 +75,20 @@ protected function initServices() {
         };
 
         $this['pdo'] = function() {
-            return new \PDO($this['config']['pdo']['dsn'], $this['config']['pdo']['username'], $this['config']['pdo']['password']);
+            $pdo = new PDO(
+                $this['config']['pdo']['dsn'],
+                $this['config']['pdo']['username'],
+                $this['config']['pdo']['password']
+            );
+            $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
```

Comment:
Have you thought about using `utf8mb4` as character set?

`$pdo->exec("SET CHARACTER SET utf8mb4");`